src/test/java/net/datafaker/service/FakerContextTest.java

```diff
@@ -0,0 +1,86 @@
+package net.datafaker.service;
+
+import net.datafaker.providers.base.BaseFaker;
+import net.datafaker.providers.base.Locality;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.util.Locale.ENGLISH;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class FakerContextTest {
+    private final RandomService random = new RandomService();
+
+    @Test
+    void localeChain() {
+        assertThat(chain(ENGLISH)).containsExactly(ENGLISH);
+        assertThat(chain(new Locale("et", "EE"))).containsExactly(new Locale("et", "EE"), new Locale("et"), new Locale("", "EE"), ENGLISH);
+        assertThat(chain(new Locale("hu_HU"))).containsExactly(new Locale("hu", "HU"), new Locale("hu"), new Locale("", "HU"), ENGLISH);
+    }
+
+    @ParameterizedTest
+    @MethodSource("languagesMatchingCountryCode")
+    void appendsDefaultCountryForGivenLanguage(String language) {
+        String countryCode = language.toUpperCase(Locale.ROOT);
+        assertThat(chain(new Locale(language))).containsExactly(new Locale(language), new Locale("", countryCode), ENGLISH);
+    }
+
+    @ParameterizedTest
+    @MethodSource("languagesDifferentFromCountryCode")
+    void appendsDefaultCountryForGivenLanguage2(String language, String country) {
```

Comment:
nit: can we rename it to have better match with what we have in `MethodSource` since suffix `2` doesn't really bring lots of benefit